### PR TITLE
soulrrrrr project proposal

### PR DESCRIPTION
**Pull request by soulrrrrr**
---
A good PR should include the following parts:

- [ ] A clear title
- [ ] A meaningful message for PR, as well as its commits
- [ ] from your specific branch (***not master***) merging to teacher's `HW` branch (***not master***, again)
- [ ] excluding any irrelevant files, such as binaries, text files, or dot files
- [ ] passing tests/CI

### CODE REVIEW

File: project/soulrrrrr/README.rst [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+
+Basic Information
+=================
+
+https://github.com/soulrrrrr/ColumnarArray
+
+Problem to Solve
+================
+
+This problem is from https://yyc.solvcon.net/en/latest/nsd/schedule/22au_nycu/schedule.html#columnar-array.
```
Comment (yungyuc) [inline project/soulrrrrr/README.rst:13]: Please reword the problem by using your own description.

File: project/soulrrrrr/README.rst [hunk 2]
```diff
@@ -0,0 +1,85 @@
...
+
+Prospective Users
+=================
+
+People originally uses Array of Struct in Python and wants to speed up.
+
+System Architecture
+===================
+
+Analyze how your system takes input, produces results, provide interface, and
```
Comment (yungyuc) [inline project/soulrrrrr/README.rst:23]: This paragraph does not seem to be about your architecture but a general notion.  It does not need to appear.

File: project/soulrrrrr/README.rst [hunk 3]
```diff
@@ -0,0 +1,85 @@
...
+performs any other operations.  Describe the system's work flow.  You may
+consider to use a flow chart but it is not required.  Specify the constraints
+assumed in your system.  Describe the modularization of the system.
+
+A class template :cpp:class:`!ColumnarArray` is holding the arrays of struct
+and the row accessor of it. The arrays of struct inside :cpp:class:`!SimpleArray`
+is :cpp:class:`!Vector`.
```
Comment (yungyuc) [inline project/soulrrrrr/README.rst:30]: If the Vector here is variable width, you need to be very careful when sharing the memory buffer between Python and C++ or there can be dangling pointers.

File: project/soulrrrrr/README.rst [hunk 4]
```diff
@@ -0,0 +1,85 @@
...
+
+At the moment I am planning, :cpp:class:`!ColumnarArray` can only deal with 
+structs have numeric types only.
```
Comment (yungyuc) [inline project/soulrrrrr/README.rst:33]: Numeric types are a good constraint to start with.

File: project/soulrrrrr/README.rst [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+4. Documentation
+
+  I'm not really sure where to publish it, I think I will write it down
+  first and then put on a github.io page.
+
+Schedule
+========
+
+* Planning phase (6 weeks from 9/19 to 10/31):
+* Week 1 (10/31): Read C++ template programming and pybind11 docs.
```
Comment (yungyuc) [inline project/soulrrrrr/README.rst:70]: It is unlikely for you to learn how to use them in one week, if you are not familiar with the constructs already.

**Review by yungyuc — CHANGES_REQUESTED** [review]
Please add your project to the list in readme file in the `project` root directory, and address all points I made.

**Comment by soulrrrrr** [comment]
Fixed the report. I didn't find the problem need to solve by columnar array, so I referred to the Pandas DataFrame class and adapted my goal.

**Comment by yungyuc** [comment]
Please follow my review comments.  After that please leave a line for me to review again.

File: project/soulrrrrr/README.rst (hunk 1 above)
Comment (soulrrrrr) [reply project/soulrrrrr/README.rst]: I added my description of this problem.

File: project/soulrrrrr/README.rst (hunk 2 above)
Comment (soulrrrrr) [reply project/soulrrrrr/README.rst]: Deleted this paragraph.

File: project/soulrrrrr/README.rst (hunk 5 above)
Comment (soulrrrrr) [reply project/soulrrrrr/README.rst:69]: I added one more week for study.

File: project/soulrrrrr/README.rst (hunk 3 above)
Comment (soulrrrrr) [reply project/soulrrrrr/README.rst]: Ok. I will implement methods to access the vector, and it cannot be directly accessed.

File: project/soulrrrrr/README.rst (hunk 4 above)
Comment (soulrrrrr) [reply project/soulrrrrr/README.rst:32]: Ok.

**Comment by soulrrrrr** [comment]
I added comment for each review. I'm not familiar with this feature of Github, please tell me if I did something wrong. Thanks.

**Comment by yungyuc** [comment]
I only see one comment from you.  You didn't respond to other comments I made and didn't follow my request in the review.  Please familiarize yourself with github and make it right.

File: project/README.rst [hunk 6]
```diff
@@ -86,4 +86,6 @@ Project List by Account Name
 #. `davidzwei <https://github.com/davidzwei>`__ for
    `KD-Tree <davidzwei/README.md>`__: https://github.com/davidzwei/KD-Tree
 #. `charles0521 <https://github.com/charles0521>`__ for
-   `Gaussian-Elimination and QR-decomposition <charles0521/README.md>`__: https://github.com/charles0521/Gaussian-Elimination-and-QR-decomposition
\ No newline at end of file
+   `Gaussian-Elimination and QR-decomposition <charles0521/README.md>`__: https://github.com/charles0521/Gaussian-Elimination-and-QR-decomposition
+#. `soulrrrrr <https://github.com/soulrrrrr>`__ for
+   `Columnar Array <soulrrrrr/README.rst>`__: https://github.com/soulrrrrr/ColumnarArray
```
Comment (soulrrrrr) [inline project/README.rst:91]: Added my project list in README.rst

**Review by soulrrrrr — COMMENTED** [review]
Added my project in README.rst.

File: project/soulrrrrr/README.rst (hunk 5 above)
Comment (yungyuc) [reply project/soulrrrrr/README.rst:69]: It's fine if you are already familiar with template.  If not, be very careful about your time management.  It's a black hole.

**Review by yungyuc — APPROVED** [review]

**Comment by yungyuc** [comment]
Let me know when you want it to be merged.

**Comment by soulrrrrr** [comment]
Solved conflict. Please merge.

**Comment by yungyuc** [comment]
Glad to see you get more familiar with github.

**Pull request merged**